### PR TITLE
[CI] Skip getting logs for long running jobs

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

For long running jobs (over ~4 hours) we are getting `Exited with status -1 (agent lost)` during Fetching Results step due to we are collecting and downloading too many logs. Buildkite agent seems to be exited automatically during this process. This PR skips downloading ray logs for those long running jobs so the test can finish successfully. 
 

## Related issue number

Closes:
- https://github.com/ray-project/ray/issues/38997
- https://github.com/ray-project/ray/issues/39053
- https://buildkite.com/ray-project/release-tests-branch/builds/2103#018a4494-7784-4da4-b2e5-e53882590420/2700
- https://buildkite.com/ray-project/release-tests-branch/builds/2103#018a4494-779b-4aae-b00a-023753d1c329/3187
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@GeneDer is this your attempt at debugging or do you actually intend to run the test for a shorter duration? The whole point of the test is to be "long running," so let's avoid shortening it

**Comment by GeneDer** [comment]
Yep, still debugging the issue for this and thus marked as draft. I think the serve job is not causing the issue. There might be something in how we log/ fetching the logs. Also seeing similar issue in some other release tests

**Comment by GeneDer** [comment]
Finally got a success run https://buildkite.com/ray-project/release-tests-pr/builds/51930#018a52b1-7c5b-4357-831a-4dbf55770f6e

**Review by can-anyscale — APPROVED** [review]
Love it, thank you for fixing! We might want to stream the last few logs for long running jobs, but I can improve that at another time.

https://media1.giphy.com/media/3oEdva9BUHPIs2SkGk/giphy.gif?cid=5a38a5a2qk7cqc49sxbdl89a8sjrohr11fsbmwh2fs7i0kkh&ep=v1_gifs_search&rid=giphy.gif&ct=g

File: release/ray_release/job_manager/anyscale_job_manager.py [hunk 1]
```diff
@@ -337,6 +338,10 @@ def get_last_logs(self):
         if self._last_logs:
             return self._last_logs
 
+        # Skip loading logs when the job ran for too long and collected too much logs.
+        if self._duration is not None and self._duration > 4 * 3_600:
+            return None
```
Comment (aslonnie) [inline release/ray_release/job_manager/anyscale_job_manager.py:343]: any unit test that can cover this?

Comment (GeneDer) [reply release/ray_release/job_manager/anyscale_job_manager.py:343]: just did 😛

**Comment by GeneDer** [comment]
@aslonnie can you help merge this if you are cool with the change? 🙏

**Review by aslonnie — APPROVED** [review]

**Pull request merged**